AEPServices/Sources/utility/hitprocessor/PersistentHitQueue.swift

```diff
@@ -63,7 +63,9 @@ public class PersistentHitQueue: HitQueuing {
             guard !self.suspended else { return }
             guard let hit = self.dataQueue.peek() else { return } // nothing left in the queue, stop processing
 
-            self.processor.processHit(entity: hit, completion: { [weak self] success in
+            
+            let semaphore = DispatchSemaphore(value: 0)
```

Comment:
need to block the current dispatchqueue until getting the respsonse

Reply:
Yeah, I can see how the bug could occur... I think this is probably the best solution, potentially using an `OperationQueue` from the Swift library could help us here but I don't think that's required.